test/unit/Loading.test.js

```diff
@@ -4,75 +4,79 @@ import {render} from 'enzyme';
             <div>Loading is done!</div>
         </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
+    expect(loading.html()).toMatchSnapshot("Loading with is_loading=true");
 });
-test('Loading renders without status', () => {
+test('Loading renders without loading_state', () => {
     const loading = render(
         <Loading>
             <div>Loading is done!</div>
         </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
+    expect(loading.html()).toEqual("<div>Loading is done!</div>");
 });
-test('Loading renders without status.isLoading', () => {
+test('Loading renders without loading_state.is_loading', () => {
     const statusMock = {
-        propName: 'children',
-        componentName: 'div'
+        prop_name: 'children',
+        component_name: 'div'
     }
     const loading = render(
-        <Loading status={statusMock}>
+        <Loading loading_state={statusMock}>
             <div>Loading is done!</div>
         </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
+    expect(loading.html()).toEqual("<div>Loading is done!</div>");
 });
-test('Loading renders without status.propName', () => {
+test('Loading renders without prop_name', () => {
     const statusMock = {
-        isLoading: true,
-        componentName: 'div'
+        is_loading: true,
+        component_name: 'div'
     }
     const loading = render(
-        <Loading status={statusMock}>
+        <Loading loading_state={statusMock}>
             <div>Loading is done!</div>
         </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
+    expect(loading.html()).toMatchSnapshot("Loading with is_loading=true");
 });
-test('Loading renders without status.componentName', () => {
+test('Loading renders without loading_state.component_name', () => {
     const statusMock = {
-        isLoading: true,
-        propName: 'children'
+        is_loading: true,
+        prop_name: 'children'
     }
     const loading = render(
-        <Loading status={statusMock}>
+        <Loading loading_state={statusMock}>
             <div>Loading is done!</div>
         </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
+    expect(loading.html()).toMatchSnapshot("Loading with is_loading=true");
 });
-test('Loading renders without children', () => {
+test('Loading renders with multiple children', () => {
     const statusMock = {
-        isLoading: false,
-        propName: 'children',
-        componentName: 'div'
+        is_loading: true,
+        prop_name: 'children',
+        component_name: 'div'
     }
     const loading = render(
-        <Loading status={statusMock} />
+        <Loading loading_state={statusMock}>
+            <div>Child 1</div>
+            <div>Child 2</div>
+            <div>Child 3</div>
+        </Loading>
     );
 
-    expect(loading.html()).toBeDefined();
-});
\ No newline at end of file
+    expect(loading.html()).toMatchSnapshot("Loading with is_loading=true");
```

Comment:
A lot better. Although I'm concerned about test stability -- updating the styling or the loading component will systematically require updating the snapshots. Is there any gain vs. simply checking if the HTML contains `dash-default-spinner`?

Reply:
The snapshots are pretty easy to update - running `npm run test-unit` and if there's a change in snapshots, will tell you to run something like `npm run test-unit -- -u` to update the snapshots. So the gain is that it's a bit easier to update if we were to change our UI. Another gain is of course that it tests if the output is exactly the way we want as opposed to testing if it has a class of `dash-default-spinner`.